File: JavaScript/overTimeNumber.js
function noOvertime(no, works) { 
	while(no>0){
		let largestNumberInWorks = Math.max.apply(Math,works);
		let largestNumberIndex = works.indexOf(largestNumberInWorks);
		works[largestNumberIndex] -= 1;
		no -= 1;
	}
	return getOverTimeNumber(works);
}

function getOverTimeNumber(works){
	var answer = 0;
	for(var i=0, item; item = works[i]; i++){
		answer += item*item;
	}
	return answer;
}

console.log(noOvertime(4,[4,3,3]))

File: Python/overTimeNumber.py
def noOvertime(n, works):
    result = 0 
    while n>0:
        works[works.index(max(works))]  -= 1
        n -= 1   
    return overTimeNumber(works)
    
    
def overTimeNumber(arr):
    answer = 0
    for i in arr:
        answer += i*i
    return answer

noOvertime(4,[4,3,3])
